Public/Src/Cache/ContentStore/Distributed/FileSystem/IFileCopier.cs

```diff
@@ -60,17 +60,19 @@ public interface ITraceableAbsolutePathFileCopier : IAbsolutePathFileCopier, ITr
     }
 
     /// <summary>
-    /// Requests another machine to copy from the current machine.
+    /// Copies files to another machine.
     /// </summary>
-    public interface ICopyRequester
+    public interface IProactiveCopier
     {
         /// <summary>
         /// Requests another machine to copy a file.
         /// </summary>
-        /// <param name="context">The context of the operation</param>
-        /// <param name="hash">The hash of the file to be copied.</param>
-        /// <param name="targetMachine">The machine that should copy the file</param>
         Task<BoolResult> RequestCopyFileAsync(OperationContext context, ContentHash hash, MachineLocation targetMachine);
+
+        /// <summary>
+        /// Pushes content to a target machine.
+        /// </summary>
+        Task<BoolResult> PushFileAsync(OperationContext context, ContentHash hash, Stream source, MachineLocation targetMachine);

```

Comment:
Cancellable?

Reply:
It is. CancellationToken is in OperationContext

---
In reply to: [343825436](https://github.com/microsoft/BuildXL/pull/1141#discussion_r343825436) [](ancestors = 343825436)